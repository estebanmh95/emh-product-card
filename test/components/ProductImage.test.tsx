import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('it', () => {
  it('renders without crashing', () => {
    const wrapperTest = renderer.create(
      <ProductImage img="http://test-image.jpg" />
    );

    expect(wrapperTest.toJSON()).toMatchSnapshot();
  });

  it('debe de mostrar el componente con la imagen del producto', () => {
    const wrapperTest = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapperTest.toJSON()).toMatchSnapshot();
  });
});
