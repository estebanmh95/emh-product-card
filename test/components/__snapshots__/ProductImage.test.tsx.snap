// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`it renders without crashing 1`] = `
<img
  alt="http://test-image.jpg"
  className="productImg undefined"
  src="http://test-image.jpg"
/>
`;
